src/backend.rs

```diff
@@ -44,7 +43,9 @@ where
 
     pub fn process(&self, u_tx: UnBuiltTransaction<Datum, Redeemer>) -> Result<()> {
         let tx = self.build(u_tx)?;
-        self.txo_record.issue(tx).map_err(|e| Error::TxORecord(e))?;
+        can_spend_inputs(&tx, self.signer().clone())
+            .map_err(|e| TxORecordError::FailedToSpendInputs(Box::new(e)))?;
```

Comment:
You don't need the `map_err` here. `can_spend_inputs` already returns the right error type.